core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java

```diff
@@ -0,0 +1,207 @@
+package org.bptlab.cepta;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.datastream.AsyncDataStream;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.bptlab.cepta.models.events.train.LiveTrainDataProtos.LiveTrainData;
+import org.bptlab.cepta.models.events.train.PlannedTrainDataProtos.PlannedTrainData;
+import org.bptlab.cepta.config.PostgresConfig;
+import org.bptlab.cepta.operators.DataToDatabase;
+import org.bptlab.cepta.operators.LivePlannedCorrelationFunction;
+import org.bptlab.cepta.operators.WeatherLocationCorrelationFunction;
+import org.bptlab.cepta.providers.LiveTrainDataProvider;
+import org.bptlab.cepta.providers.PlannedTrainDataProvider;
+import org.bptlab.cepta.providers.WeatherDataProvider;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import com.google.protobuf.GeneratedMessage;
+
+import sun.awt.image.SunWritableRaster.DataStealer;
+
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.sql.*;
+
+public class DataToDatabaseTests {
+
+    public void initDatabaseStuff(PostgreSQLContainer container) {
+        // CREATE TABLE
+        String db_url = container.getJdbcUrl();
+        String user = container.getUsername();
+        String password = container.getPassword();
+        
+        Connection conn = null;
+        Statement stmt = null;
+        try{
+           // Register JDBC driver
+           Class.forName("org.postgresql.Driver");
+     
+           // Open a connection
+           System.out.println("Connecting to a database...");
+           conn = DriverManager.getConnection(db_url, user, password);
+           System.out.println("Connected database successfully...");
+           
+           stmt = conn.createStatement();
+           String sql;
+           // Create table for planned data
+           sql = createPlannedDatabaseQuery();
+           stmt.executeUpdate(sql);
+     
+        }catch(SQLException se){
+           //Handle errors for JDBC
+           se.printStackTrace();
+        }catch(Exception e){
+           //Handle errors for Class.forName
+           e.printStackTrace();
+        }finally{
+           //finally block used to close resources
+           try{
+              if(stmt!=null)
+                 conn.close();
+           }catch(SQLException se){
+           }// do nothing
+           try{
+              if(conn!=null)
+                 conn.close();
+           }catch(SQLException se){
+              se.printStackTrace();
+           }//end finally try
+        }//end try
+        System.out.println("Goodbye!");
+   }
+public int checkDatabaseInput(PostgreSQLContainer container) {
```

Comment:
Basically the same remarks as in initDatabaseStuff. Maybe outsource the connection/disconnection to the database..

Reply:
did it!